darts/timeseries.py

```diff
@@ -307,47 +307,56 @@ def from_series(pd_series: pd.Series,
 
     @staticmethod
     def from_times_and_values(times: Union[pd.DatetimeIndex, pd.RangeIndex],
-                              values: Union[np.ndarray, pd.DataFrame],
+                              values: np.ndarray,
                               fill_missing_dates: Optional[bool] = True,
                               freq: Optional[str] = None,
                               columns: Optional[pd._typing.Axes] = None) -> 'TimeSeries':
         """
-        Returns a deterministic TimeSeries built from an index and values.
+        Returns a TimeSeries built from an index and value array.
 
         Parameters
         ----------
         times
             A `pandas.DateTimeIndex` or `pandas.RangeIndex` representing the time axis for the time series.
         values
-            A 2-dimensional Numpy array of values for the TimeSeries. The dimensions should be (time, component).
+            A Numpy array of values for the TimeSeries. Both 2-dimensional arrays, for deterministic series,
+            and 3-dimensional arrays, for probabilistic series, are accepted. In the former case the dimensions
+            should be (time, component), and in the latter case (time, component, sample).
         fill_missing_dates
             Optionally, a boolean value indicating whether to fill missing dates with NaN values. This requires
             either a provided `freq` or the possibility to infer the frequency from the provided timestamps.
         freq
             Optionally, a string representing the frequency of the Pandas DataFrame. This is useful in order to fill
             in missing values if some dates are missing and `fill_missing_dates` is set to `True`.
-        columns
-            Columns to be used by the underlying pandas DataFrame.
 
         Returns
         -------
         TimeSeries
             A TimeSeries constructed from the inputs.
         """
 
-        # TODO: make it more general, not always going via a DataFrame
         raise_if_not(isinstance(times, pd.RangeIndex) or isinstance(times, pd.DatetimeIndex),
                      'the `times` argument must be a RangeIndex or a DateTimeIndex. Use '
                      'TimeSeries.from_values() if you want to use an automatic RangeIndex.')
-        df = pd.DataFrame(values, index=times)
+
+        if not times.name:
```

Comment:
Sound logic, however, are you maybe mutating the time index provided in argument here?
You could maybe simply store a `time_name` variable here.